lago/config.py

```diff
@@ -14,88 +14,216 @@
+                    env_lago[section][param] = value
+                except ValueError:
+                    warn(
+                        'warning - malformed environment variable name:'
+                        '{0}, ignoring.'.format(
+                            key
+                        )
+                    )
+    return dict(env_lago)
+
+
+class ConfigLoad(object):
+    """Merges configuration parameters from 3 different sources:
+    1. Enviornment vairables
+    2. config files in .INI format
+    3. argparse.ArgumentParser
+
+    The assumed order(but not necessary) order of calls is:
+    load() - load from config files and environment variables
+    update_parser(parser) - update from the declared argparse parser
+    update_args(args) - update from passed arguments to the parser
+    """
+
+    def __init__(self, root_section='lago'):
+        """__init__
+        Args:
+            root_section (str):
+        """
+
+        self.root_section = root_section
+        self._config = defaultdict(dict)
+        self._config.update(self.load())
+        self._parser = None
+
+    def load(self):
+        """Load all configuration from INI format files and ENV, always
+        preferring the last read.
+        Order of loading is:
+            1) Custom paths as defined in constants.CONFS_PATH
+            2) XDG standard paths
+            3) Environment variables
+
+        Returns:
+            dict: dict of section configuration dicts
+
+        """
+
+        configp = configparser.ConfigParser()
+        for path in _get_configs_path():
+            try:
+                with open(path, 'r') as config_file:
+                    configp.read_file(config_file)
+            except IOError:
+                pass
+        configp.read_dict(get_env_dict(self.root_section))
+        return {s: dict(configp.items(s)) for s in configp.sections()}
+
+    def update_args(self, args):
+        """Update config dictionary with parsed args, as resolved by argparse.
+        Only root positional arguments that already exist will overridden.
+
+        Args:
+            args (namespace): args parsed by argparse
+        """
+
+        for arg in vars(args):
+            if self.get(arg):
+                self._config[self.root_section][arg] = getattr(args, arg)
+
+    def update_parser(self, parser):
+        """Update config dictionary with declared arguments in an argparse.parser
+        New variables will be created, and existing ones overridden.
+
+        Args:
+            parser (argparse.ArgumentParser): parser to read variables from
+
+        """
+
+        self._parser = parser
+        ini_str = argparse_to_ini(parser)
+        configp = configparser.ConfigParser(allow_no_value=True)
+        configp.read_dict(self._config)
+        configp.read_string(ini_str)
+        self._config.update(
+            {
+                s: dict(configp.items(s))
+                for s in configp.sections()
+            }
```

Comment:
Good point, would probably fail(don't have 2.6 env running here to test), we don't support it right?(in the docs I see 2.7)